src/templates/BlogPostTemplate/blogPostTemplate.tsx

```diff
@@ -0,0 +1,117 @@
+const BlogPostTemplate = ({
+  topic,
+  title,
+  author,
+  date,
+  headerImgUrl,
+  headerImgAlt,
+  slices,
+}: BlogPostTemplateProps) => {
+  // returns either date or author, or if both are given "date - author"
+  const getDateAuthorString = (): string | null => {
+    if (!date && !author) {
+      return null;
+    }
+    if (date) {
+      const formattedDate = new Date(date).toLocaleDateString("en-US", {
+        year: "numeric",
+        month: "long",
+        day: "numeric",
+      });
+      return author ? `${formattedDate} - ${author}` : formattedDate;
+    }
+    return author || null;
+  };
+
+  const dateAuthorString = getDateAuthorString();
+
+  return (
+    <div>
+      <div>
+        {topic && <p>{topic}</p>}
+        {title && <h1>{title}</h1>}
+        {dateAuthorString && <p>{dateAuthorString}</p>}
+      </div>
+      {headerImgUrl && headerImgAlt && (
+        <ImageBlock url={headerImgUrl} caption={headerImgAlt} isFullWidth />
+      )}
+      {/* eslint-disable @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-return */}
+      {slices &&
+        slices.length > 0 &&
+        slices.map((slice) => {
+          switch (slice.slice_type) {
+            case "text_block":
+              return <TextBlock rawText={slice.primary.body_text.raw} />;
+            case "file_download_block":
+              return (
+                <ButtonBlock
+                  header={slice.primary.file_download_header.text}
+                  url={slice.primary.file.url}
+                  buttonText={slice.primary.button_text.text}
+                  isExternalLink
+                />
+              );
+            case "quote_block":
+              return (
+                <QuoteBlock
+                  quote={slice.primary.quote.text}
+                  attributee={slice.primary.attributee.text}
+                />
+              );
+            case "stats_block":
+              return (
+                <StatsBlock
+                  statistic={slice.primary.statistic.text}
+                  statistic_text={slice.primary.statistic_text.text}
+                />
+              );
+            case "image_with_caption":
+              return (
+                slice.primary.image.url &&
+                slice.primary.caption.text && (
+                  <ImageBlock
+                    url={slice.primary.image.url}
+                    caption={slice.primary.caption.text}
+                  />
+                )
+              );
+            case "cta_block":
+              return (
+                <ButtonBlock
+                  header={slice.primary.header.text}
+                  url={slice.primary.button_link.url}
+                  buttonText={slice.primary.button_text.text}
+                />
+              );
+            case "separator":
+              return (
+                <img src={LogoSeparator} alt="Divider with ShelterTech logo" />
```

Comment:
It's generally not useful to add alt text to presentation-only images like this, since it's not helpful to people using screen readers. The [W3C official recommendation](https://www.w3.org/QA/Tips/altAttribute) is to set `alt=""` for images like this:

> if the image is used to create bullets in a list, a horizontal line, or other similar decoration, it is fine to have an empty alt attribute (e.g., alt=""), but it is better to use things like list-style-image in CSS

To provide a motivating example, imagine if you sent a blog post through a text-to-speech program, and between paragraphs, it suddenly spoke "Divider with ShelterTech logo."